xcube/webapi/ows/coverages/controllers.py

```diff
@@ -144,34 +190,270 @@ def get_coverage_data(
+
+    return _IndexerTuple(indices, slices, x_value, y_value)
+
+
+def _apply_geographic_subsetting(
+    ds, subset_crs, indexers
+) -> tuple[list[float], xr.Dataset]:
+    # NB: We use xy axis ordering for the bounding boxes throughout this
+    # function, regardless of what's specified in the CRSs.
+
+    # 1. transform native extent to a whole-dataset bbox in subset_crs.
+    # We'll use this to fill in "full extent" values if geographic
+    # subsetting is only specified in one dimension.
+    full_bbox_native = get_bbox_from_ds(ds)
+    native_crs = get_crs_from_dataset(ds)
+    full_bbox_subset_crs = _transform_bbox(
+        full_bbox_native, native_crs, subset_crs
+    )
+
+    # 2. Find horizontal and/or vertical ranges in indexers, falling back to
+    # values from whole-dataset bbox if a complete bbox is not specified.
+    x0, x1 = (
+        indexers.x
+        if indexers.x is not None
+        else (full_bbox_subset_crs[0], full_bbox_subset_crs[2])
+    )
+    y0, y1 = (
+        indexers.y
+        if indexers.y is not None
+        else (full_bbox_subset_crs[1], full_bbox_subset_crs[3])
+    )
+
+    # 3. Using the ranges determined from the indexers and whole-dataset bbox,
+    # construct the requested bbox in the subsetting CRS.
+    bbox_subset_crs = [x0, y0, x1, y1]
+
+    # 4. Transform requested bbox from subsetting CRS to dataset-native CRS.
+    bbox_native_crs = _transform_bbox(bbox_subset_crs, subset_crs, native_crs)
+
+    # 6. Apply the dataset-native bbox using sel, making sure that y/latitude
+    # slice has the same ordering as the corresponding co-ordinate.
+    h_dim = _get_h_dim(ds)
+    v_dim = _get_v_dim(ds)
+    ds = ds.sel(
+        indexers={
+            h_dim: slice(bbox_native_crs[0], bbox_native_crs[2]),
+            v_dim: slice(
+                *_correct_inverted_y_range(
+                    ds, v_dim, (bbox_native_crs[1], bbox_native_crs[3])
+                )
+            ),
+        }
+    )
+    return bbox_subset_crs, ds
+
+
+def get_bbox_from_ds(ds: xr.Dataset):
+    h, v = ds[_get_h_dim(ds)], ds[_get_v_dim(ds)]
+    bbox = list(map(float, [h[0], v[0], h[-1], v[-1]]))
+    _ensure_bbox_y_ascending(bbox)
+    return bbox
+
+
+def _find_geographic_parameters(
+    names: list[str],
+) -> tuple[Optional[str], Optional[str]]:
+    x, y = None, None
+    for name in names:
+        if name.lower()[:3] in ['x', 'e', 'eas', 'lon']:
+            x = name
+        if name.lower()[:3] in ['y', 'n', 'nor', 'lat']:
+            y = name
+    return x, y
+
+
+def _transform_bbox(
+    bbox: list[float], source_crs: pyproj.CRS, dest_crs: pyproj.CRS
+) -> list[float]:
+    if source_crs == dest_crs:
+        return bbox
+    else:
+        transformer = pyproj.Transformer.from_crs(
+            source_crs, dest_crs, always_xy=True
+        )
+        bbox_ = bbox.copy()
+        _ensure_bbox_y_ascending(bbox_)
+        return list(transformer.transform_bounds(*bbox_))
```

Comment:
You can remove line 379 and move everything up one indent :)

Reply:
A matter of taste, but OK, done :)